include/unifex/await_transform.hpp

```diff
@@ -112,6 +113,14 @@ struct _awaitable_base<Promise, Value>::type {
       result_->state_ = _state::exception;
       continuation_.resume();
     }
+
+    template (class Error)
+      (requires is_exception_ptr_convertible_v<Error>)
+    void set_error(Error &&error) && noexcept {
+      unifex::activate_union_member(result_->exception_, get_exception_ptr(std::forward<Error>(error)));
+      result_->state_ = _state::exception;
+      continuation_.resume();
+    }
```

Comment:
If you widen the contract of `get_exception_ptr` as described below, then you only need one (unconstrained) `set_error` here that unconditionally passes the argument through `get_exception_ptr`.